mrp_subcontracting_skip_no_negative/models/stock_move.py

```diff
@@ -69,10 +69,10 @@ def _action_done(self, cancel_backorder=False):
                             location=location.complete_name,
                         )
                     )
-        res = super(StockMove, self - moves_with_no_check)._action_done(
+        res = super(StockMove, moves_with_no_check)._action_done(
             cancel_backorder=cancel_backorder
         )
-        res += super(StockMove, moves_with_no_check)._action_done(
+        res += super(StockMove, self - moves_with_no_check)._action_done(
             cancel_backorder=cancel_backorder
         )
```

Comment:
Correct me if I'm wrong, but I doubt that the proposed implementation works well when the receipt picking contains subcontracted and normal products, with a partial quantities processed for the normal product (i.e., there should be a backorder which contain both products for the remaining quantities). I think the current logic will generate two backorder pickings instead of one.

Instead, would it perhaps work if we simply process `_action_done()` once for all moves?

```py
        moves_with_check = self - moves_with_no_check
        res = super(
            StockMove, moves_with_check + moves_with_no_check
        )._action_done(cancel_backorder=cancel_backorder)
```

Reply:
> Instead, would it perhaps work if we simply process _action_done() once for all moves?

@yostashiro We just need to pass context for subcontract moves to skip the check. 
So, processing all moves at once will not be worked well.

Reply:
> We just need to pass context for subcontract moves to skip the check.
> So, processing all moves at once will not be worked well.

As we talked, context can only be passed for the recordset.
Can't pass different context for the record of recordset.

Reply:
> I think the current logic will generate two backorder pickings instead of one.

No, It will just create only one backorder.